### PR TITLE
FEATURE: BB-1388 Add support to group rows for attribute columns

### PR DESCRIPTION
**Pull request by martinmosko**

### CODE REVIEW

**Comment by gdgate** [comment]
Build in pipeline check aborted.

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/3082/ : SUCCESS in 6s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/2437/ : FAILURE in 3m 54s

**Comment by martinmosko** [comment]
extended test - storybook

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/3083/ : SUCCESS in 10s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/2438/ : SUCCESS in 6m 54s

**Comment by yenkins** [comment]
Link to Storybook test result:

https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-storybook-zuul-docker/1279/artifact/test-storybook/html-report/index.html

**Comment by gdgate** [comment]
Build failed (check-extended pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/3084/ : SUCCESS in 10s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-pull-request-dispatcher/761/ : UNSTABLE in 29m 22s

**Comment by martinmosko** [comment]
extended test - storybook

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/3086/ : SUCCESS in 12s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/2440/ : SUCCESS in 6m 41s

**Comment by gdgate** [comment]
Build in pipeline check-extended aborted.

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/3089/ : SUCCESS in 6s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/2441/ : SUCCESS in 2m 27s

**Comment by martinmosko** [comment]
extended test - storybook

**Comment by yenkins** [comment]
Link to Storybook test result:

https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-storybook-zuul-docker/1282/artifact/test-storybook/html-report/index.html

**Comment by gdgate** [comment]
Build succeeded (check-extended pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/3090/ : SUCCESS in 8s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-pull-request-dispatcher/764/ : SUCCESS in 21m 31s

File: src/components/core/PivotTable.tsx [hunk 1]
```diff
@@ -311,6 +314,11 @@ export const getAGGridDataSource = (
                     }
                 );
                 const { offset, count, total } = execution.executionResult.paging;
+
+                const rowAttributeIds = columnDefs
+                    .filter(columnDef => columnDef.type === 'ROW_ATTRIBUTE_COLUMN')
```
Comment (petrjanu) [inline src/components/core/PivotTable.tsx]: Use constant `ROW_ATTRIBUTE_COLUMN`

File: styles/scss/pivotTable.scss [hunk 2]
```diff
@@ -153,6 +153,20 @@ $header-cell-resize-width: 20px;
...
         }
     }
+
+    .gd-column-merged {
+        display: none;
+    }
+
+    .gd-cell-hide {
+        color: transparent;
+        border-top: 1px solid white;
```
Comment (petrjanu) [inline styles/scss/pivotTable.scss]: Will `translarent` color work?

File: styles/scss/pivotTable.scss [hunk 3]
```diff
@@ -153,6 +153,20 @@ $header-cell-resize-width: 20px;
...
+        background-color: white;
```
Comment (petrjanu) [inline styles/scss/pivotTable.scss]: Do we need to override bg with white background?

File: src/components/core/PivotTable.tsx [hunk 4]
```diff
@@ -401,12 +409,14 @@ export class PivotTableInner extends BaseVisualization<IPivotTableInnerProps, IP
         updateTotals: noop,
         onDataTooLarge: noop,
         pageSize: 100,
-        config: {}
+        config: {},
+        groupRows: false
```
Comment (petrjanu) [inline src/components/core/PivotTable.tsx]: Do we need default value here? Maybe default value in parent component is enough.

File: src/components/core/tests/PivotTable.spec.tsx [hunk 5]
```diff
@@ -271,6 +280,24 @@ describe('PivotTable', () => {
             ]);
         });
     });
+
+    describe('groupRows for attribute columns', () => {
+        it.each([
```
Comment (petrjanu) [inline src/components/core/tests/PivotTable.spec.tsx]: 👍

File: src/components/core/PivotTable.tsx (hunk 4 above)
Comment (martinmosko) [reply src/components/core/PivotTable.tsx]: we have two PivotTable components:
- Buckets PivotTable -> this is public interface for UI.SDK
- Core PivotTable -> this is used directly by our AD/KD 

I would say it's should be tested on both places as we are using directly this core component :)

Comment (petrjanu) [reply src/components/core/PivotTable.tsx]: Ah, okay

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/3118/ : SUCCESS in 7s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/2468/ : SUCCESS in 2m 59s

File: styles/scss/pivotTable.scss (hunk 2 above)
Comment (martinmosko) [reply styles/scss/pivotTable.scss]: it looks like it could be :)

File: styles/scss/pivotTable.scss (hunk 3 above)
Comment (martinmosko) [reply styles/scss/pivotTable.scss]: this is required to fulfil UX design requirement

**Comment by gdgate** [comment]
Build in pipeline check aborted.

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/3122/ : SUCCESS in 9s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/2471/ : SUCCESS in 6m 19s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/3126/ : SUCCESS in 11s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/2475/ : SUCCESS in 14m 47s

**Comment by petrjanu** [comment]
LGTM

**Comment by gdgate** [comment]
Build succeeded (gate pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-merge-compliance-check/671/ : SUCCESS in 20s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/2476/ : SUCCESS in 27m 38s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-storybook-zuul-docker/1300/ : SUCCESS in 25m 19s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-examples-zuul-docker/653/ : SUCCESS in 26m 30s

**Pull request merged**